### PR TITLE
HPCC-15404 Optimize roxie standalone queries

### PR DESCRIPTION
**Pull request by richardkchapman**

### CODE REVIEW

**Comment by richardkchapman** [comment]
@ghalliday Please review/merge

File: roxie/ccd/ccdqueue.cpp [hunk 1]
```diff
@@ -746,6 +778,11 @@ class RoxieQueue : public CInterface
             queueLength++;
             if (queueLength>maxQueueLength)
                 maxQueueLength = queueLength;
+            if (idle==0 && started < numWorkers)
```
Comment (ghalliday) [inline roxie/ccd/ccdqueue.cpp]: There is a minor race condition here.  If there is 1 idle thread, and multiple enqueue()s occur at the same time, no new threads will be started.

File: roxie/ccd/ccdqueue.cpp [hunk 2]
```diff
@@ -1332,6 +1371,21 @@ class CRoxieWorker : public CInterface, implements IPooledThread
...
 };
 
+IPooledThread *RoxieQueue::createNew()
+{
+    return new CRoxieWorker;
+}
+
+void RoxieQueue::abortChannel(unsigned channel)
+{
+    Owned<IPooledThreadIterator> wi = workers->running();
```
Comment (ghalliday) [inline roxie/ccd/ccdqueue.cpp]: Does this need protection inside a critical section?

**Comment by ghalliday** [comment]
@richardkchapman a couple of threading questions

File: roxie/ccd/ccdqueue.cpp (hunk 1 above)
Comment (richardkchapman) [inline roxie/ccd/ccdqueue.cpp]: I suspect avoidance of race conditions was the motivation for pre-starting the threads in the first place.

File: roxie/ccd/ccdqueue.cpp (hunk 2 above)
Comment (richardkchapman) [inline roxie/ccd/ccdqueue.cpp]: Possibly, though I suspect no more so than it did previously.

**Comment by richardkchapman** [comment]
@ghalliday Back to you

**Pull request merged**